sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/InMemoryCatalog.scala

```diff
@@ -48,7 +48,7 @@ class InMemoryCatalog extends ExternalCatalog {
   private val catalog = new scala.collection.mutable.HashMap[String, DatabaseDesc]
 
   private def filterPattern(names: Seq[String], pattern: String): Seq[String] = {
-    val regex = pattern.replaceAll("\\*", ".*").r
+    val regex = pattern.replaceAll("\\.\\*", "\\*").replaceAll("\\*", "\\.\\*").r
```

Comment:
Looks like Hive's doc does not really match its implementation. I also took a look at postgres (http://www.postgresql.org/docs/9.1/static/app-psql.html#APP-PSQL-PATTERNS) and impala (http://www.cloudera.com/documentation/archive/impala/2-x/2-1-x/topics/impala_show.html#show_functions_unique_1). `*` will be interpreted as `.*` in regex. Also, looking at our current implementation, we do not really have a super clear syntax for the pattern. I feel it will be better to decide the syntax first and then make the change.